tasks/main.yml

```diff
@@ -48,3 +48,10 @@
   command: >
     /bin/chown -Rv solr:solr "{{ solr_core_conf_dest }}"
   when: sync_solr_conf.changed
+
+- name: Restart solr
+  cron:
+    name: Restart Solr at 8 am UTC
+    job: service solr restart
```

Comment:
A few notes: 
- Cron tab usually doesn't come with any "envrionment"; not even $PATH. So you you need to specify the absolute paths to binaries.  Also, since this role can be used on both redhat/debian, you'll need to set up an os-dependent variable for the path to service.
- The name of the service isn't necessarily named "solr" - you need to use the variable that the role provides.
- The service command sometimes issues boring "stopped / started" messages to stdout. You  want to redirect non-error output to `> /dev/null` to keep noise down. Errors will still come to stderr.

Reply:
I would also:
- createa a default (that defaults to true) that lets us choose whether we actually want this feature or not, and have the cron job be present or absent based on that variable,
- let the hour + minute be configurable

Reply:
>Cron tab usually doesn't come with any "envrionment"; not even $PATH. So you you need to specify the absolute paths to binaries. Also, since this role can be used on both redhat/debian, you'll need to set up an os-dependent variable for the path to service.
By absolute path to binaries, you mean instead of service, I should try like /bon/solr restat?

I tried this on redhat and seems like service solr restart works for it as well. So do we need a os-dependant variable here.

>The name of the service isn't necessarily named "solr" - you need to use the variable that the role provides.
The service command sometimes issues boring "stopped / started" messages to stdout. You want to redirect non-error output to > /dev/null to keep noise down. Errors will still come to stderr.

Done

>createa a default (that defaults to true) that lets us choose whether we actually want this feature or not, and have the cron job be present or absent based on that variable,

Done

let the hour + minute be configurable

Done

Reply:
> I tried this on redhat

You'll have to explain to me what you tried, because on a stock CentOS 7 box, with a cron job of:
```
$ sudo crontab -l

* * * * * service crond status
```
I get this: 
```
$ sudo tail /var/spool/mail/root 

X-Cron-Env: <SHELL=/bin/sh>
X-Cron-Env: <HOME=/root>
X-Cron-Env: <PATH=/usr/bin:/bin>
X-Cron-Env: <LOGNAME=root>
X-Cron-Env: <USER=root>
Message-Id: <20201214215701.A39C1453B1E@localhost>
Date: Mon, 14 Dec 2020 21:57:01 +0000 (UTC)

/bin/sh: service: command not found
```
PATH does not include `/sbin`, which is where the service command lives, so the command fails.

Trust me. Always wiring full paths to binaries in cron jobs is worth doing the first time. You should always assume an inconsistent or non-existent environment in cron tabs, except for those env vars that you wire right into your cron job line. This is the voice of bitter experience speaking.

The rest of this is looking good so far.

Reply:
@dale-c-anderson : Done.

The sole restart handler  fails for u16.04. https://github.com/AcroMedia/ansible-role-solr-core-config/runs/1554874755?check_suite_focus=true

Could be because of https://github.com/ansible/ansible/pull/72702? Because when I comment out the "state"restarted" in the handler, the play works.

Reply:
Yeah it's something to do with that. It could be that Solr 8 + Ubuntu 16.04 are a poor match. I notice Geerlingguy's solr role doesn't test that combination. 

I say we either: 
- Update our matrix so we can specify both the OS + Solr version to use (probably not worth the effort for this role), which would let us avoid solr 8 on ubuntu 18, or 
- Just drop Ubuntu 16.04 from this role's test matrix. This role isn't really going to work any differently on different OS versions; as long as we get it passing on one flavor of ubuntu + centos, that's fine.